### PR TITLE
[Merged by Bors] - chore(Mathlib/Init/ZeroOne): rename `Mathlib.Init.ZeroOne` to `Mathlib.Algebra.Group.ZeroOne`

### PR DESCRIPTION
**Pull request by Komyyy**
---

- [x] depends on: #14750

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Komyyy** [comment]
> Sorry, I'll veto the `Order.Defs` change. I'm working on a bigger rework and it would help not having weird diffs because someone moved the file.

I see, @YaelDillies.

File: scripts/noshake.json [hunk 1]
```diff
@@ -224,15 +224,15 @@
   "Mathlib.Tactic.Nontriviality.Core": ["Mathlib.Logic.Nontrivial.Basic"],
   "Mathlib.Tactic.NoncommRing": ["Mathlib.Algebra.Group.Action.Defs"],
   "Mathlib.Tactic.MoveAdd":
-  ["Mathlib.Algebra.Group.Basic", "Mathlib.Init.Order.LinearOrder"],
+  ["Mathlib.Algebra.Group.Basic"],
```
Comment (Ruben-VandeVelde) [inline scripts/noshake.json:227]:
```suggestion
  ["Mathlib.Algebra.Group.Basic", "Mathlib.Order.Defs"],
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by YaelDillies** [comment]
Can you remove the `Order.Defs` changes, then?

**Comment by Komyyy** [comment]
@YaelDillies Yes.

File: Mathlib/Algebra/ZeroOne.lean [hunk 2]
```diff
@@ -7,17 +7,6 @@ import Mathlib.Tactic.ToAdditive
 import Mathlib.Mathport.Rename
 
 /-!
-# Note about `Mathlib/Init/`
```
Comment (jcommelin) [inline Mathlib/Algebra/ZeroOne.lean:10]: Please move the last 5 lines in this file to a better home.

File: scripts/noshake.json (hunk 1 above)
Comment (Komyyy) [reply scripts/noshake.json]: This change is reverted.

**Comment by jcommelin** [comment]
You seem to also be moving the `to_additive` frontend to a new file. Is that intentional?

**Comment by Komyyy** [comment]
@jcommelin Yes, this PR moves `attribute [to_additive]` commands in `Mathlib.Init.ZeroOne` to better home.

**Comment by YaelDillies** [comment]
The diff is sadly messed up. Can you make sure that git understands your `to_additive` changes as moves?

**Comment by Komyyy** [comment]
@YaelDillies
I rebased the commit to make git recognize this change as a rename. But the diff didn't change.

File: Mathlib/Tactic/ToAdditive.lean [hunk 3]
```diff
@@ -1,1496 +1,15 @@
...
-import Mathlib.Lean.Meta.Simp
-import Lean.Elab.Tactic.Ext
-import Lean.Meta.Tactic.Symm
-import Lean.Meta.Tactic.Rfl
-import Batteries.Lean.NameMapAttribute
-import Batteries.Tactic.Lint -- useful to lint this file and for for DiscrTree.elements
-import Mathlib.Tactic.Relation.Trans -- just to copy the attribute
-import Mathlib.Tactic.Eqns -- just to copy the attribute
-import Mathlib.Tactic.Simps.Basic
+import Mathlib.Tactic.ToAdditive.Frontend
```
Comment (jcommelin) [inline Mathlib/Tactic/ToAdditive.lean:6]: Can you please undo the changes to this file. Maybe this is a good change, but it should be a separate PR.

Comment (Komyyy) [reply Mathlib/Tactic/ToAdditive.lean:6]: #14750

**Comment by YaelDillies** [comment]
> @YaelDillies I rebased the commit to make git recognize this change as a rename. But the diff didn't change.

Doesn't matter how the commits are *within your PR*. What matters is how the commits are *across PRs*. You might have to split your PR into two/change the way you rename the files

**Comment by Komyyy** [comment]
@YaelDillies
OK.
Fortunately, another PR was in the process of merging this change as a split PR, so I cancelled it.
I will re-split this change tomorrow as it is already late today Japan time.

File: Mathlib/Algebra/ZeroOne.lean
Comment (YaelDillies) [inline Mathlib/Algebra/ZeroOne.lean:1]: Could it be renamed to `Algebra.Group.ZeroOne`?

Comment (Komyyy) [reply Mathlib/Algebra/ZeroOne.lean:1]: I renamed.

**Comment by j-loreaux** [comment]
Thanks!

bors merge
